tests/ui/as_pointer_underscore.rs

```diff
@@ -0,0 +1,17 @@
+#![warn(clippy::as_pointer_underscore)]
+#![crate_type = "lib"]
+#![no_std]
+
+struct S;
+
+fn f(s: &S) -> usize {
+    &s as *const _ as usize
+    //~^ ERROR: using inferred pointer cast
+    //FIXME: idk how to get this help working ~| HELP: use explicit type: `*const &S`
```

Comment:
How it should be done?

Reply:
I mean, the help message is printed, but I can't get it to work with `~| HELP: ...` matching here specifically. Sorry for not being precise

Reply:
Sorry, just looked at the new code now. Are you referring to the formatting of the message? You'll have to use `span_lint_and_then` and the `span_suggestion_with_style`. I think we'll be changing the default suggestion style at some point in the future.